File: Backend/src/EventifyDomain/Speaker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EventifyDomain
{
    public class Speaker
    {
        public int Id { get; init; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string ImageURL { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public IEnumerable<SocialMedia> SocialMedia { get; set; }

        public IEnumerable<SpeakerEvent> SpeakerEvents { get; set; }
    }
}

File: Frontend/EventifyApp/src/app/models/Speaker.ts
import { Event } from './Event';
import { SocialMedia } from "./SocialMedia";

export interface Speaker {
    id: number;
    name: string;
    description: string;
    imageURL: string;
    phone: string;
    email: string;
    socialMedia: SocialMedia[];
    speakerEvents: Event[];
}
